### PR TITLE
Learninglocker

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

We want to use learning locker in our stack and save in it xAPI statements.
To be able to deploy it we want to configure it in Arnold like we did for OpenEDX, marsha or richie.

## Proposal

Learning locker requires learning locker application, xapi-service, mongodb, redis, nginx and mailcatcher (only in non production environment).

Learning locker application has itself 3 applications to run, I group then in only one (the `app` service in arnold) service to ease its configuration, they share the exact same stack, this is why there are several `DeploymentConfig` and `Service` objects in this service.

Learning locker is also the first application in arnold having `ConfigMaps` to be consumed as environment variables in a `DeploymentConfig`.

- [x] Create learning locker application, divided in multiple services described before.
- [x] Activate and configure learning locker for eugene customer.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 
 - Create ConfigMaps to be consumed in environment variables in a DeploymentConfig.
+- Add learning locker in the list of configured application.
```
Comment (jmaupetit) [inline CHANGELOG.md]:
```suggestion
- Add LearningLocker to the list of supported applications.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+- Activate and configure learning locker for eugene customer in development environment.
```
Comment (jmaupetit) [inline CHANGELOG.md]:
```suggestion
- Activate and configure LearningLocker for the eugene customer in development & CI environments.
```

File: apps/learninglocker/templates/app/_env.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,8 @@
+API_PORT: "{{ learninglocker_api_port }}"
+UI_PORT: "{{ learninglocker_ui_port }}"
+LOG_MIN_LEVEL: info
+LOG_DIR: /tmp
+NODE_ENV: production
+WINSTON_CLOUDWATCH_ENABLED: false
```
Comment (jmaupetit) [inline apps/learninglocker/templates/app/_env.yml.j2:10]: What does this variable stand for?

File: apps/learninglocker/templates/app/dc_worker.yml.j2 [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{% set service_variant = "worker" %}
+{% set learninglocker_command = "worker/dist/server" %}
+{% set learninglocker_port = "8080" %}
```
Comment (jmaupetit) [inline apps/learninglocker/templates/app/dc_worker.yml.j2]: This shoudn't be an integer and configurable?

File: apps/learninglocker/templates/app/secret.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app: learninglocker
+    service: "api-ui-worker"
```
Comment (jmaupetit) [inline apps/learninglocker/templates/app/secret.yml.j2]: As it is required for all learning locker services, can we remove this label?

File: apps/learninglocker/templates/mongodb/svc.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  namespace: "{{ project_name }}"
+spec:
+  ports:
+  - name: "{{ learninglocker_mongodb_port }}-tcp"
+    port: {{ learninglocker_mongodb_port }}
+    protocol: TCP
+    targetPort: {{ learninglocker_mongodb_port }}
+# As commented in the ad hoc endpoint, the endpoint name points to this service
+# so that it does not rely on a deployment configuration when the "env_type" is
+# not trashable. In this case, we use a PostgreSQL cluster outside of
```
Comment (jmaupetit) [inline apps/learninglocker/templates/mongodb/svc.yml.j2]:
```suggestion
# not trashable. In this case, we use a mongodb cluster outside of
```

File: apps/learninglocker/templates/nginx/_configs/default.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  server learninglocker-xapi:{{ learninglocker_xapi_port }};
+}
+
+upstream backend_api {
+  server learninglocker-api:{{ learninglocker_api_port }};
+}
+
+server {
+  listen       {{ learninglocker_nginx_port }} default_server;
+  listen       [::]:{{ learninglocker_nginx_port }} default_server;
```
Comment (jmaupetit) [inline apps/learninglocker/templates/nginx/_configs/default.conf.j2:16]: Can you add a comment to explain this please? I don't get it.

Comment (lunika) [reply apps/learninglocker/templates/nginx/_configs/default.conf.j2:16]: It's for binding IPV6 addresses

File: apps/learninglocker/templates/app/dc_worker.yml.j2 (hunk 4 above)
Comment (lunika) [reply apps/learninglocker/templates/app/dc_worker.yml.j2]: it should not have listening port. It a copy past fail.

File: apps/learninglocker/templates/app/_env.yml.j2 (hunk 3 above)
Comment (lunika) [reply apps/learninglocker/templates/app/_env.yml.j2:10]: to enable loggin on cloudwatch. Should I put a link to all available environment variables ?

Comment (jmaupetit) [reply apps/learninglocker/templates/app/_env.yml.j2:10]: Maybe yeah.

File: apps/learninglocker/templates/app/dc_worker.yml.j2 (hunk 4 above)
Comment (jmaupetit) [reply apps/learninglocker/templates/app/dc_worker.yml.j2]: IC

File: apps/learninglocker/templates/nginx/_configs/default.conf.j2 (hunk 7 above)
Comment (jmaupetit) [reply apps/learninglocker/templates/nginx/_configs/default.conf.j2:16]: Okay then. Add a comment please.

File: apps/learninglocker/templates/app/_env.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{#  see every env variables available http://docs.learninglocker.net/guides-configuring/#learning-locker-application #}
```
Comment (jmaupetit) [inline apps/learninglocker/templates/app/_env.yml.j2]:
```suggestion
{#  
  An exhaustive list of learning locker environment variables is available at:
  http://docs.learninglocker.net/guides-configuring/#learning-locker-application 
#}
```

File: apps/learninglocker/templates/nginx/_configs/default.conf.j2 [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  server learninglocker-xapi:{{ learninglocker_xapi_port }};
+}
+
+upstream backend_api {
+  server learninglocker-api:{{ learninglocker_api_port }};
+}
+
+server {
+  listen       {{ learninglocker_nginx_port }} default_server;
+  # listen all IPV6 adresses. Like line before for IPV4.
```
Comment (jmaupetit) [inline apps/learninglocker/templates/nginx/_configs/default.conf.j2]:
```suggestion
  # Also listen to IPV6 requests
```

Comment (lunika) [reply apps/learninglocker/templates/nginx/_configs/default.conf.j2]: done

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**